query/stdlib/influxdata/influxdb/from.go

```diff
@@ -6,107 +6,18 @@ import (
 	"github.com/influxdata/flux"
 	"github.com/influxdata/flux/codes"
 	"github.com/influxdata/flux/plan"
-	"github.com/influxdata/flux/runtime"
-	"github.com/influxdata/flux/semantic"
 	"github.com/influxdata/flux/stdlib/influxdata/influxdb"
-	platform "github.com/influxdata/influxdb"
 )
 
 const FromKind = "influxDBFrom"
 
-type FromOpSpec struct {
-	Bucket   string `json:"bucket,omitempty"`
-	BucketID string `json:"bucketID,omitempty"`
-}
-
-func init() {
-	fromSignature := semantic.MustLookupBuiltinType("influxdata/influxdb", "from")
-	runtime.ReplacePackageValue("influxdata/influxdb", influxdb.FromKind, flux.MustValue(flux.FunctionValue(FromKind, createFromOpSpec, fromSignature)))
-	flux.RegisterOpSpec(FromKind, newFromOp)
-	plan.RegisterProcedureSpec(FromKind, newFromProcedure, FromKind)
-}
-
-func createFromOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
-	spec := new(FromOpSpec)
-
-	if bucket, ok, err := args.GetString("bucket"); err != nil {
-		return nil, err
-	} else if ok {
-		spec.Bucket = bucket
-	}
-
-	if bucketID, ok, err := args.GetString("bucketID"); err != nil {
-		return nil, err
-	} else if ok {
-		spec.BucketID = bucketID
-	}
-
-	if spec.Bucket == "" && spec.BucketID == "" {
-		return nil, &flux.Error{
-			Code: codes.Invalid,
-			Msg:  "must specify one of bucket or bucketID",
-		}
-	}
-	if spec.Bucket != "" && spec.BucketID != "" {
-		return nil, &flux.Error{
-			Code: codes.Invalid,
-			Msg:  "must specify only one of bucket or bucketID",
-		}
-	}
-	return spec, nil
-}
-
-func newFromOp() flux.OperationSpec {
-	return new(FromOpSpec)
-}
-
-func (s *FromOpSpec) Kind() flux.OperationKind {
-	return FromKind
-}
-
-// BucketsAccessed makes FromOpSpec a query.BucketAwareOperationSpec
-func (s *FromOpSpec) BucketsAccessed(orgID *platform.ID) (readBuckets, writeBuckets []platform.BucketFilter) {
-	bf := platform.BucketFilter{}
-	if s.Bucket != "" {
-		bf.Name = &s.Bucket
-	}
-	if orgID != nil {
-		bf.OrganizationID = orgID
-	}
-
-	if len(s.BucketID) > 0 {
-		if id, err := platform.IDFromString(s.BucketID); err != nil {
-			invalidID := platform.InvalidID()
-			bf.ID = &invalidID
-		} else {
-			bf.ID = id
-		}
-	}
-
-	if bf.ID != nil || bf.Name != nil {
-		readBuckets = append(readBuckets, bf)
-	}
-	return readBuckets, writeBuckets
-}
+type (
+	NameOrID   = influxdb.NameOrID
+	FromOpSpec = influxdb.FromOpSpec
+)
 
 type FromProcedureSpec struct {
```

Comment:
Maybe this should be renamed `FromStorageProcedureSpec` to differentiate it from the the one in the `flux` repo?  I'm finding the differences are getting confusing.

Reply:
I'll do that rename. I think I'm also going to update the rule to convert to the storage procedure spec if host is unset at all and add a validation to ensure that org isn't set to the storage procedure spec. I think the error messages will get confusing otherwise.